### PR TITLE
Doc/block graph

### PR DESCRIPTION
**Pull request by AureliaDolo**

### CODE REVIEW

**Review by yvan-sraka — DISMISSED** [review]

**Review by yvan-sraka — APPROVED** [review]

**Review by yvan-sraka — APPROVED** [review]

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors cancel

**Comment by AureliaDolo** [comment]
bors cancel

File: consensus/src/block_graph.rs [hunk 1]
```diff
@@ -51,36 +51,63 @@ impl HeaderOrBlock {
...
     pub block: Block,
-    pub parents: Vec<(BlockId, u64)>, // one (hash, period) per thread ( if not genesis )
-    pub children: Vec<BlockHashMap<u64>>, // one HashMap<hash, period> per thread (blocks that need to be kept)
-    pub dependencies: BlockHashSet,       // dependencies required for validity check
+    /// one (block id, period) per thread ( if not genesis )
+    pub parents: Vec<(BlockId, u64)>,
+    /// one HashMap<Block id, period> per thread (blocks that need to be kept)
+    /// Children reference that block as a parent
+    pub children: Vec<BlockHashMap<u64>>,
+    /// dependencies required for validity check (TODO: not sure if it's used)
```
Comment (AureliaDolo) [inline consensus/src/block_graph.rs:91]: @damip are we using dependencies once the block is active ?

File: consensus/src/block_graph.rs [hunk 2]
```diff
@@ -90,15 +117,26 @@ impl ActiveBlock {
...
```
Comment (AureliaDolo) [inline consensus/src/block_graph.rs:131]: same here

File: consensus/src/block_graph.rs (hunk 1 above)
Comment (damip) [reply consensus/src/block_graph.rs:91]: yes, for graph pruning, to know which other blocks it depends on

File: consensus/src/block_graph.rs (hunk 2 above)
Comment (damip) [reply consensus/src/block_graph.rs:131]: used for graph pruning (see previous answer)

**Review by damip — DISMISSED** [review]
LGTM, just remove the TODOs on descendants

**Comment by AureliaDolo** [comment]
bors merge

**Comment by AureliaDolo** [comment]
bors merge

**Pull request merged**